lib/Sema/CSGen.cpp

```diff
@@ -1916,61 +1917,39 @@ namespace {
           options |= TypeResolutionFlags::AllowPackReferences;
           elementEnv = OuterExpansions.back();
         }
-        const auto result = TypeResolution::resolveContextualType(
+        auto result = TypeResolution::resolveContextualType(
             specializationArg, CurDC, options,
             // Introduce type variables for unbound generics.
             OpenUnboundGenericType(CS, locator),
             HandlePlaceholderType(CS, locator),
             OpenPackElementType(CS, locator, elementEnv));
-        if (result->hasError())
-          return true;
-
+        if (result->hasError()) {
+          auto &ctxt = CS.getASTContext();
+          auto *repr = new (ctxt) PlaceholderTypeRepr(specializationArg->getLoc());
+          result = PlaceholderType::get(ctxt, repr);
```

Comment:
Instead of create a placeholder repr placeholder type should use original `specializationArg`

Reply:
Fixed

Reply:
It looks like `replaceInferableTypesWithTypeVars` does this same pattern of creating a `PlaceholderTypeRepr`, so I think this should be left as is.

Reply:
That method is actually going the other way - it detects if there are placeholders in a `Type` and "opens" them. I think we should expand from PlaceholderTypeRepr to `TypeRepr` in `PlaceholderType` and locator element. But we can do that in a separate PR.